### PR TITLE
Removed API call for culprit if job passes and code refactoring

### PR DESCRIPTION
**Pull request by monishaChengappa**

### CODE REVIEW

File: groupStatusesWriter.php [hunk 1]
```diff
@@ -1,119 +1,75 @@
 <?php
 
-    function get_values(){
-      $params = parse_ini_file('config');
-      return array_values($params);
+class JenkinsConfig
+{
```
Comment (taoistmath) [inline groupStatusesWriter.php]: This brace should be on previous line

File: groupStatusesWriter.php [hunk 2]
```diff
@@ -1,119 +1,75 @@
...
+            exit('Could not get xml');
         }
-        return false;
     }
 
-    function get_culprit($jobName) {      
-      if($xml = simplexml_load_file('https://'.get_user().':'.get_pass().'@'.get_site().'/job/'.$jobName.'/lastFailedBuild/api/xml')) {
+    public function getAllJobsPath() {
+        return 'json/allJobs.json';
+    }
```
Comment (taoistmath) [inline groupStatusesWriter.php]: Does it make sense to have a function which returns a string?

File: groupStatusesWriter.php [hunk 3]
```diff
@@ -1,119 +1,75 @@
...
+      if($xml = simplexml_load_file('https://'.$con->get_user().':'.$con->get_pass().'@'.$con->get_site().'/job/'.$jobName.'/lastFailedBuild/api/xml')) {
           if(isset($xml->culprit->fullName)){
             $culprit = $xml->culprit->fullName;
           } else {
-            $culprit = "No culprit";
+            $culprit = "No culprit returned";
           }
       } else {
-          $culprit = $jobName." appears to be in good shape.";
+          $culprit = "No last fail";
```
Comment (taoistmath) [inline groupStatusesWriter.php]: how about 'No failed build returned'? to me this sounds better, but open for debate

**Comment by taoistmath** [comment]
I believe each class should be it's own file which is then imported into groupStatusWriter

File: groupStatusesWriter.php [hunk 4]
```diff
@@ -1,119 +1,75 @@
...
-
-    function prior_dump($jobName) {
-        if(prior_check($jobName)){
-            remove_fail($jobName);
+    
+    public function getStatusXml(){
+        if (file_exists('results/status.xml')) {
+            return simplexml_load_file('results/status.xml');
+        } else {
+            exit('Failed to open results/status.xml');
```
Comment (dutchb) [inline groupStatusesWriter.php]: Message says failed to open, logic says doesn't exist.

**Comment by dutchb** [comment]
:+1:

**Comment by taoistmath** [comment]
+1

**Pull request merged**